tracer_options.go

```diff
@@ -95,6 +95,14 @@ func (tracerOptions) Extractor(format interface{}, extractor Extractor) TracerOp
 }
 
 func (tracerOptions) Observer(observer Observer) TracerOption {
+	old := oldObserver{obs: observer}
+
+	return func(tracer *Tracer) {
+		tracer.observer.append(&old)
+	}
```

Comment:
sorry, yes delegate to scoped tracerOptions.ContribObserver function, so there's no naming conflicts.